pyanaconda/modules/boss/kickstart_manager.py

```diff
@@ -0,0 +1,139 @@
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+from pydbus.error import map_error
+from pydbus.auto_names import auto_object_path
+
+from pyanaconda.dbus import DBus
+from pyanaconda.dbus.constants import DBUS_BOSS_ANACONDA_NAME
+
+from pyanaconda.kickstart_dispatcher.parser import SplitKickstartParser, VALID_SECTIONS_ANACONDA
+from pykickstart.version import makeVersion
+from pykickstart.errors import KickstartError, KickstartParseError
+
+from pyanaconda.anaconda_loggers import get_module_logger
+log = get_module_logger(__name__)
+
+
+__all__ = ['KickstartManager', 'SplitKickstartError']
+
+
+@map_error("{}.SplitKickstartError".format(DBUS_BOSS_ANACONDA_NAME))
+class SplitKickstartError(Exception):
+    """Error when parsing kickstart for splitting."""
+    pass
+
+
+class SplitKickstartUnknownSectionError(SplitKickstartError):
+    """Unknown section was found in kickstart."""
+    pass
+
+
+class SplitKickstartMissingIncludeError(SplitKickstartError):
+    """File included in kickstart was not found."""
+    pass
+
+
+class KickstartManager(object):
+    """Distibutes kickstart to modules and collects it back."""
+
+    def __init__(self):
+        self._kickstart_path = None
+        self._elements = None
+        self._module_observers = []
+
+    @property
+    def module_observers(self):
+        """Get all module observers for kickstart distribution."""
+
+        return self._module_observers
+
+    @module_observers.setter
+    def module_observers(self, modules):
+        """Set module observers for kicstart distribution.
+
+        :param modules: Module observers list
+        :type modules: list(DBusObjectObserver)
+        """
+        self._module_observers = modules
+
+    @property
+    def elements(self):
+        """Return all elements of split kickstart."""
+        return self._elements
+
+    @property
+    def unprocessed_kickstart(self):
+        """Return kickstart not processed by any module."""
+        return self._elements.get_kickstart_from_elements(self._elements.unprocessed_elements)
+
+    def split(self, path):
+        """Split the kickstart given by path into elements."""
+        self._elements = None
+        self._kickstart_path = path
+        handler = makeVersion()
+        ksparser = SplitKickstartParser(handler, valid_sections=VALID_SECTIONS_ANACONDA)
+        try:
+            result = ksparser.split(path)
+        except KickstartParseError as e:
```

Comment:
Actually, you can map the` KickstartParseError` and `KickstartError` to a DBus error if you want to.